src/runtimes/wapc/evaluation_context_registry.rs

```diff
@@ -0,0 +1,329 @@
+/// policies being instantiated. This is all modelled to optimize the Policy Server scenario,
+/// although everything works fine also with kwctl.
+///
+///
+/// ## The problem
+///
+/// In the Policy Server scenario, we have multiple workers. Each one of them living inside of
+/// their own dedicated thread.
+///
+/// Given a unique list of waPC policies, a worker will instantiate only one wapc runtime per
+/// policy. This is done to reduce the amount of memory consumed by Policy Server.
+/// Later, when inside of the [`host_callback`](crate::runtimes::wapc::callback::host_callback) function,
+/// we receive a `wapc_id` and the details about an operation that has to be run by the Wasm
+/// host. When that happens, we need to access some auxiliary information about the policy being
+/// evaluated. For example, if a policy is requesting "list all the Kubernetes Secret objects
+/// inside of the `kube-system` Namespace", we need to know if the administrator granted access to
+/// Kubernetes Secret to the policy.
+/// The auxiliary information are stored inside of a `EvaluationContext` object.
+///
+/// ## The data structures
+///
+/// This module defines two global variables that hold all the information required by `host_callback` to
+/// obtain the `EvaluationContext` instance associated with a waPC policy. All of that by just
+/// doing a series of lookups based on the `wapc_id` associated with the policy.
+///
+/// The `WAPC_ID_TO_WORKER_ID` structure contains the relationship waPC policy -> worker.
+/// Given a waPC policy ID, we can discover to which worker it belongs.
+/// Then, using the `WORKER_ID_TO_CTX` structure, we find the `EvaluationContext` associated with a
+/// certain worker.
+///
+/// ## Workflow
+///
+/// ### Policy registration
+///
+/// As soon as a waPC policy is created, the following information have to be inserted into the
+/// registry:
+///   - wapc ID
+///   - ID of the worker that owns the policy
+///   - The first `EvaluationContext` to be used
+///
+/// ### Validate request/settings
+///
+/// Prior to invoking the `validate`/`validate_settings` functions exposed by a policy, the worker
+/// must inform the registry about the `EvaluationContext` that it's going to be used during the
+/// evaluation. This ensures the `host_callback`, if ever called by policy, obtains the right
+/// auxiliary information.
+use anyhow::{anyhow, Result};
+use lazy_static::lazy_static;
+use std::{
+    collections::HashMap,
+    sync::{
+        atomic::{AtomicU64, Ordering},
+        Arc, RwLock,
+    },
+};
+use tracing::debug;
+
+use crate::evaluation_context::EvaluationContext;
+
+lazy_static! {
+    /// A map with wapc_id as key, and the worker_id as value. It allows us to know to which
+    /// worker a waPC policy belongs. When inside of the
+    /// [`host_callback`](crate::runtimes::wapc::callback::host_callback) function,
+    /// we need to know
```

Comment:
this last phrase sounds cut. "need to know the EvaluationContext" perhaps?

Reply:
thanks, I've fixed it